web/src/hooks/useSettings.ts

```diff
@@ -169,25 +170,39 @@ type ChatSettingsParams = {
   changeSetting: (path: string[], value: any) => void;
 };
 
+type SettingsUpdatePair = [path: string[], val: any];
+
 export default function useSettings() {
   const [settingsLoaded, setLoaded] = useState(false);
   const [settings, updateSettings] = useState<LLMSettings>(makeSettings({}));
   const router = useRouter();
 
-  const updateInUrl = (path: string[], value: any) =>
+  const updateInUrl = (vals: { [key: string]: any }) =>
     router.replace({
       pathname: router.pathname,
-      query: {
-        ...router.query,
-        [path.join(".")]: value.toString(),
-      },
+      query: { ...router.query, ...vals },
     });
 
   const changeSetting = (path: string[], value: any) => {
-    updateInUrl(path, value);
+    updateInUrl({ [path.join(".")]: value });
     updateSettings((settings) => ({ ...updateIn(settings, path, value) }));
   };
 
+  const changeSettings = (...items: SettingsUpdatePair) => {
+    updateInUrl(
+      items.reduce(
+        (acc, [path, val]) => ({ ...acc, [path.join(".")]: val }),
+        {}
+      )
+    );
+    updateSettings((settings) =>
+      items.reduce(
+        (acc, [path, val]) => ({ ...acc, ...updateIn(settings, path, val) }),
+        settings
+      )
+    );
```

Comment:
not in this PR, but the local state settings could not be derived from URL? sounds like asking for trouble trying to synchronize 2 places on write instead of deriving state from 1 source of truth 🤔